xarray/backends/api.py

```diff
@@ -414,6 +415,56 @@ def _dataset_from_backend_dataset(
     return ds
 
 
+def _datatree_from_backend_datatree(
+    backend_tree,
+    filename_or_obj,
+    engine,
+    chunks,
+    cache,
+    overwrite_encoded_chunks,
+    inline_array,
+    chunked_array_type,
+    from_array_kwargs,
+    **extra_tokens,
+):
+    if not isinstance(chunks, int | dict) and chunks not in {None, "auto"}:
+        raise ValueError(
+            f"chunks must be an int, dict, 'auto', or None. Instead found {chunks}."
+        )
+
+    # _protect_datatree_variables_inplace(backend_tree, cache)
```

Comment:
@shoyer, @TomNicholas (or anyone else), do you know if we need this? Once again this is copied from `open_dataset`, but I can't quite figure out what this is guarding against (my tests appear to work without it).

Reply:
I'm also not sure, but is this guarding against xarray opening an array that could in theory be modified (e.g. `zarr.Array`) and modifying it on disk? i.e. it ensures that `xarray.open_dataset`'s only ever reads files, doesn't modify them? It would be nice if `CopyOnWriteArray` actually had a docstring...

I say we copy this across for now and raise an issue to discuss adding a test that would actually catch this.

Reply:
cc @dcherian

Reply:
> is this guarding against xarray opening an array that could in theory be modified (e.g. zarr.Array) and modifying it on disk

Yes IIRC